### PR TITLE
Remove deprecation Bruker, EMD, Nexus

### LINKED ISSUE #24: Remove deprecated code
A number of readers have deprecation warnings, usually for specific keywords -- sometimes HyperSpy v2.0 is explicitly mentioned as the point where they will be removed. We could consider to remove these deprecated keywords already before releasing the first version of RosettaSciIO as it would be a good occasion to do so and the library should be first used by HyperSpy 2.0 anyway.

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change
Remove deprecated keywords and attributes from readers before first release (which will be the one used by HyperSpy 2.0, for which these deprecations were announced).

Closes #24

### Progress of the PR
- [X] Change implemented (can be split into several points),
- [X] update user guide (if appropriate),
- [X] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/rosettasciio/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [X] remove tests,
- [X] ready for review.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -8,6 +8,20 @@ https://rosettasciio.readthedocs.io/en/latest/changes.html
 
 .. towncrier release notes start
 
+API Removal
+-----------
```
Comment (ericpre) [inline CHANGES.rst:12]: As this is relevant to hyperpsy users and the API was in hyperspy before, would it be better to move it to the corresponding entry of hyperpsy 2.0 changelog?

File: rsciio/tests/test_nexus_hdf.py [hunk 2]
```diff
@@ -26,9 +26,8 @@
 import traits.api as t
 import h5py
 
-from hyperspy.exceptions import VisibleDeprecationWarning as HSVisibleDeprecationWarning
+from hyperspy.exceptions import VisibleDeprecationWarning as VisibleDeprecationWarning
```
Comment (ericpre) [inline rsciio/tests/test_nexus_hdf.py:29]:
```suggestion
from hyperspy.exceptions import VisibleDeprecationWarning
```

File: CHANGES.rst (hunk 1 above)
Comment (jlaehne) [reply CHANGES.rst:12]: Good point, we might want to just duplicate it there, in case migrating HyperSpy users end up looking here?

Comment (jlaehne) [reply CHANGES.rst:12]: see https://github.com/hyperspy/hyperspy/pull/3002

Comment (ericpre) [reply CHANGES.rst:12]: I still find odd to have a `API removal` section for a first release! I would remove this section and a normal changelog entry - the description on its own is good.

File: rsciio/tests/test_nexus_hdf.py (hunk 2 above)
Comment (ericpre) [reply rsciio/tests/test_nexus_hdf.py:29]: What's about this?

Comment (jlaehne) [reply rsciio/tests/test_nexus_hdf.py]: I commited it online, but then overwrote it when pushing the updated version from my PC ;-) Should be fixed now.

File: CHANGES.rst (hunk 1 above)
Comment (jlaehne) [reply CHANGES.rst]: I moved it to a regular towncrier entry for the PR and checked the formatting locally.

**Review by ericpre — APPROVED** [review]
Thanks!

**Pull request merged**